test/driftexport/drift_export_test.go

```diff
@@ -0,0 +1,733 @@
+}
+
+func createFilter(parent *record, table *schema.Table, resource string, q squirrel.SelectBuilder) (squirrel.SelectBuilder, error) {
+	if info, ok := exportedTables[resource]; ok && info.sqlPred != nil {
+		q = q.Where(info.sqlPred)
+	}
+	if parent != nil {
+		id, ok := parent.data["cq_id"]
+		if !ok {
+			return q, fmt.Errorf("cq_id field is missing in a parent of %s", table.Name)
+		}
+		var fk string
+		for _, col := range table.Columns {
+			if strings.HasSuffix(col.Name, "_cq_id") {
+				fk = col.Name
+				break
+			}
+		}
+		if fk == "" {
+			return q, fmt.Errorf("foreign key not found for the table %s", table.Name)
+		}
+		q = q.Where(squirrel.Eq{fk: id})
+	}
+	return q, nil
+}
+
+func loadTable(ctx context.Context, conn *pgxpool.Conn, table *schema.Table, resource string, parent *record) ([]record, error) {
+	var result []record
+	psql := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar)
+	q := psql.Select(fmt.Sprintf("to_jsonb(%s)", table.Name)).From(table.Name)
+	q, err := createFilter(parent, table, resource, q)
+	if err != nil {
+		return nil, err
+	}
+	sql, args, err := q.ToSql()
+	if err != nil {
+		return nil, err
+	}
+	rows, err := conn.Query(ctx, sql, args...)
+	if err != nil {
+		return nil, err
+	}
+	for rows.Next() {
+		var row record
+		if err := rows.Scan(&row.data); err != nil {
+			return nil, err
+		}
+		if isInterestingRecord(row.data) {
+			result = append(result, row)
+		}
+	}
+	if err := rows.Err(); err != nil {
+		return nil, err
+	}
+	for i, row := range result {
+		row.children = make(map[string][]record)
+		for _, relation := range table.Relations {
+			var err error
+			row.children[relation.Name], err = loadTable(ctx, conn, relation, relation.Name, &row)
+			if err != nil {
+				return nil, err
+			}
+		}
+		result[i] = row
+	}
+	return result, nil
+}
+
+func isInterestingRecord(row map[string]interface{}) bool {
+	tagsValue, ok := row["tags"]
+	if !ok {
+		return true
+	}
+	tags, ok := tagsValue.(map[string]interface{})
+	if !ok {
+		return true
+	}
+	return reflect.DeepEqual(tags["Type"], "integration_test") && reflect.DeepEqual(tags["TestId"], testSuffix)
+}
+
+func makeIDBuilders(resourceMap map[string]*schema.Table) map[string]func([]map[string]interface{}) string {
+	b := make(map[string]func([]map[string]interface{}) string)
+	for name, table := range resourceMap {
+		info, ok := exportedTables[name]
+		if ok {
+			b[name] = makeIDBuilder(info.idFormat, table)
+		}
+		children := make(map[string]*schema.Table)
+		for _, t := range table.Relations {
+			children[t.Name] = t
+		}
+		for name, f := range makeIDBuilders(children) {
+			b[name] = f
+		}
+	}
+	return b
+}
+
+func makeIDBuilder(expr string, table *schema.Table) func([]map[string]interface{}) string {
+	if expr == "" {
+		keys := make([]string, 0, len(table.Options.PrimaryKeys))
+		for _, k := range table.Options.PrimaryKeys {
+			if k != "account_id" && k != "region" && k != "cq_id" && !strings.HasSuffix(k, "_cq_id") {
```

Comment:
You could use the `column.Meta().Resolver.Name` ...

Reply:
Well, yeah, I've seen how it's done in drift code, but this was quicker.

Shamefully adopted some of the code from there. Thanks.